bundles/org.openhab.binding.ventaair/src/main/java/org/openhab/binding/ventaair/internal/VentaThingHandler.java

```diff
@@ -0,0 +1,335 @@
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.ventaair.internal.message.DeviceInfoMessage;
+import org.openhab.binding.ventaair.internal.message.Header;
+import org.openhab.binding.ventaair.internal.message.Info;
+import org.openhab.binding.ventaair.internal.message.Measurements;
+import org.openhab.binding.ventaair.internal.message.action.Action;
+import org.openhab.binding.ventaair.internal.message.action.AllActions;
+import org.openhab.binding.ventaair.internal.message.action.AutomaticAction;
+import org.openhab.binding.ventaair.internal.message.action.BoostAction;
+import org.openhab.binding.ventaair.internal.message.action.ChildLockAction;
+import org.openhab.binding.ventaair.internal.message.action.FanAction;
+import org.openhab.binding.ventaair.internal.message.action.HumidityAction;
+import org.openhab.binding.ventaair.internal.message.action.PowerAction;
+import org.openhab.binding.ventaair.internal.message.action.SleepModeAction;
+import org.openhab.binding.ventaair.internal.message.action.TimerAction;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.unit.ImperialUnits;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link VentaThingHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Stefan Triller - Initial contribution
+ */
+@NonNullByDefault
+public class VentaThingHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(VentaThingHandler.class);
+
+    @NonNullByDefault({})
+    private VentaAirDeviceConfiguration config;
+
+    private @Nullable Communicator communicator;
+
+    private Map<String, State> channelValueCache = new HashMap<>();
+
+    public VentaThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Handle command={} for channel={} with channelID={}", command, channelUID, channelUID.getId());
+        if (command instanceof RefreshType) {
+            refreshChannelFromCache(channelUID);
+            return;
+        }
+
+        switch (channelUID.getId()) {
+            case VentaAirBindingConstants.CHANNEL_POWER:
+                if (command instanceof OnOffType) {
+                    OnOffType power = (OnOffType) command;
+                    dispatchActionToDevice(new PowerAction(power == OnOffType.ON ? true : false));
```

Comment:
If you do a comparison only, you don't need to cast the command and neither check the type.

Same for below.
```suggestion
                    dispatchActionToDevice(new PowerAction(command == OnOffType.ON));
```